parseSchema/index.js

```diff
@@ -0,0 +1,111 @@
+"use strict";
+
+const Parse = require('parse/node');
+
+function updateSchemaOnlyWhenNecessary(existingSchemaData, intendedSchema)
+{
+  // remove any existing fields from the request.
+  Object.getOwnPropertyNames(existingSchemaData.fields).forEach(existingField => { delete intendedSchema._fields[existingField] ; });
```

Comment:
this method of comparison does not allow us to change the type of a field. What if we want to change a boolean to a string? I tried locally and the change is ignore. Can we do a deeper comparison between objects?

Reply:
I think it's better to avoid going into that situation in the first place...  Otherwise we would have to set rules to convert existing values of a field from one data type (which could be of custom type) to another...

Reply:
Allow me to explain further.  A request to add field is just that... to add one or more fields that do not exist yet.  Such requests can't be used to change data type of a field if I read the documentation correctly.  I assume an attempt to add an existing field but of a different type, will be rejected just as an attempt to add an existing field of the same type.